js/views/message_detail_view.js

```diff
@@ -89,6 +89,23 @@
 
             this.listenTo(this.model, 'change', this.render);
         },
+        events: {
+            'click button.delete': 'onDelete'
+        },
+        onDelete: function() {
+            var dialog = new Whisper.ConfirmationDialogView({
+                message: i18n('deleteWarning'),
+                okText: i18n('delete'),
+                resolve: function() {
+                    this.model.destroy();
+                    this.resetPanel();
+                }.bind(this),
+                reject: function() {}
```

Comment:
No-op reject is a bit arcane. I can see that this is a side-effect of having the cancel button's presence depend on the presence of a reject handler function.

Optional chore/refactor: consider either adding explicit option to ConfirmationDialogView like `hideCancel:  true` or maybe even a separate view to use when you only want to put up a notice with a dismissal, as opposed to one where you want to fork behavior based on a user choice.